crates/e2e/src/xts.rs

```diff
@@ -211,7 +211,8 @@ where
             .unwrap_or_else(|err| {
                 panic!("error on ws request `contracts_instantiate`: {:?}", err);
             });
-        scale::Decode::decode(&mut bytes.as_ref()).expect("decoding failed")
+        scale::Decode::decode(&mut bytes.as_ref())
+            .unwrap_or_else(|err| panic!("decoding failed: {}", err))
```

Comment:
Would be useful to know what failed to decode e.g. `"decoding ContractInstantiateResult failed`. Same for other `"decoding failed"` messages

Reply:
sure thing, fixed in the follow up commit